core/sync/chan/chan.odin

```diff
@@ -36,31 +82,167 @@ Raw_Chan :: struct {
+	}
+*/
 create :: proc{
 	create_unbuffered,
 	create_buffered,
 }
 
+/*
+Creates an unbuffered version of the specified `Chan` type.
+
+*Allocates Using Provided Allocator*
+
+**Inputs**
+- `$C`: Type of `Chan` to create
+- `allocator`: The allocator to use
+
+**Returns**:
+- The initialized `Chan`
+- An `Allocator_Error`
+
+Example:
+
+	import "core:sync/chan"
+
+	create_unbuffered_example :: proc() {
+		c, err := chan.create_unbuffered(chan.Chan(int), context.allocator)
+		assert(err == .None)
+		defer chan.destroy(c)
+	}
+*/
 @(require_results)
 create_unbuffered :: proc($C: typeid/Chan($T), allocator: runtime.Allocator) -> (c: C, err: runtime.Allocator_Error)
 	where size_of(T) <= int(max(u16)) {
 	c.impl, err = create_raw_unbuffered(size_of(T), align_of(T), allocator)
 	return
 }
 
+/*
+Creates a buffered version of the specified `Chan` type.
+
+*Allocates Using Provided Allocator*
+
+**Inputs**
+- `$C`: Type of `Chan` to create
+- `cap`: The capacity of the channel
+- `allocator`: The allocator to use
+
+**Returns**:
+- The initialized `Chan`
+- An `Allocator_Error`
+
+Example:
+
+	create_buffered_example :: proc() {
+		c, err := chan.create_buffered(chan.Chan(int), 10, context.allocator)
+		assert(err == .None)
+		defer chan.destroy(c)
+	}
+*/
 @(require_results)
 create_buffered :: proc($C: typeid/Chan($T), #any_int cap: int, allocator: runtime.Allocator) -> (c: C, err: runtime.Allocator_Error)
 	where size_of(T) <= int(max(u16)) {
 	c.impl, err = create_raw_buffered(size_of(T), align_of(T), cap, allocator)
 	return
 }
 
+/*
+Creates a buffered or unbuffered `Raw_Chan` for messages of the specified
+size and alignment.
+
+*Allocates Using Provided Allocator*
+
+**Inputs**
+- `msg_size`: The size of the messages the messages being sent
+- `msg_alignment`: The alignment of the messages being sent
+- [`cap`: The capacity of the channel] omit for creating unbuffered channels
+- `allocator`: The allocator to use
+
+**Returns**:
+- The initialized `Raw_Chan`
+- An `Allocator_Error`
+
+Example:
+
+	import "core:sync/chan"
+
+	create_raw_example :: proc() {
+		unbuffered: ^chan.Raw_Chan
+		buffered: ^chan.Raw_Chan
+		err: runtime.Allocator_Error
+
+		unbuffered, err = chan.create_raw(size_of(int), align_of(int), context.allocator)
+		assert(err == .None)
+		defer chan.destroy(unbuffered)
+
+		buffered, err = chan.create_raw(size_of(int), align_of(int), 10, context.allocator)
+		assert(err == .None)
+		defer chan.destroy(buffered)
+	}
+*/
 create_raw :: proc{
 	create_raw_unbuffered,
 	create_raw_buffered,
 }
 
+/*
+Creates an unbuffered `Raw_Chan` for messages of the specified
+size and alignment.
+
+*Allocates Using Provided Allocator*
+
+**Inputs**
+- `msg_size`: The size of the messages the messages being sent
+- `msg_alignment`: The alignment of the messages being sent
+- `allocator`: The allocator to use
+
+**Returns**:
+- The initialized `Raw_Chan`
+- An `Allocator_Error`
+
+Example:
+
+	import "core:sync/chan"
+	import "base:runtime"
```

Comment:
Unused import.

Reply:
Also there are a couple of examples where the import statement isn't properly indented, e.g.:
```odin
  import "core:sync/chan" // indented with spaces
	try_recv_example :: proc() {
```